src/test/java/teammates/test/cases/browsertests/FeedbackMcqQuestionUiTest.java

```diff
@@ -291,4 +292,177 @@ public void testDeleteQuestionAction() {
+
+        ______TS("MCQ: Weight cells are hidden by default");
+        feedbackEditPage.clickAddQuestionButton();
+        feedbackEditPage.selectNewQuestionTypeAndWaitForNewQuestionPanelReady("MCQ");
+        // Check if the 'Choices are weighted' checkbox unchecked by default.
+        assertFalse(feedbackEditPage.isMcqAssignWeightCheckboxChecked(NEW_QUESTION_INDEX));
+        // Check if the MCQ weights column hidden by default
+        assertFalse(feedbackEditPage.getMcqWeightsColumn(NEW_QUESTION_INDEX).isDisplayed());
+        assertFalse(feedbackEditPage.getMcqOtherWeightBox(NEW_QUESTION_INDEX).isDisplayed());
+
+        ______TS("MCQ: Weight cells are visible when weights are assigned");
+        feedbackEditPage.clickMcqAssignWeightCheckboxForNewQuestion();
+        // Check if the MCQ weights column are visible or not.
+        assertTrue(feedbackEditPage.getMcqWeightsColumn(NEW_QUESTION_INDEX).isDisplayed());
+        // Check the 'other' option is disabled and otherWeight cell is hidden
+        assertFalse(browser.driver.findElement(By.id("mcqOtherOptionFlag-" + NEW_QUESTION_INDEX)).isSelected());
+        assertFalse(feedbackEditPage.getMcqOtherWeightBox(NEW_QUESTION_INDEX).isDisplayed());
+
+        ______TS("MCQ: Other weight Cell is visible when other option and weights both are enabled");
+
+        // Check Assign MCQ weights Enabled but other option disabled
+        assertTrue(feedbackEditPage.isMcqAssignWeightCheckboxChecked(NEW_QUESTION_INDEX));
+        assertFalse(browser.driver.findElement(By.id("mcqOtherOptionFlag-" + NEW_QUESTION_INDEX)).isSelected());
+
+        // Assign Other option and test that other weight cell is displayed.
+        feedbackEditPage.clickAddMcqOtherOptionCheckboxForNewQuestion();
+        assertTrue(feedbackEditPage.getMcqOtherWeightBox(NEW_QUESTION_INDEX).isDisplayed());
+
+        ______TS("MCQ: Check Mcq weight default value");
+        feedbackEditPage.fillQuestionTextBox("MCQ weight feature", NEW_QUESTION_INDEX);
+        feedbackEditPage.fillQuestionDescription("More details", NEW_QUESTION_INDEX);
+
+        // Fill MCQ choices and check corresponding weight values
+        feedbackEditPage.fillMcqOptionForNewQuestion(0, "Choice 1");
+        feedbackEditPage.verifyFieldValue("mcqWeight-0--1", "0");
+        feedbackEditPage.fillMcqOptionForNewQuestion(1, "Choice 2");
+        feedbackEditPage.verifyFieldValue("mcqWeight-1--1", "0");
+
+        // Verify MCQ other weight default value
+        feedbackEditPage.verifyFieldValue("mcqOtherWeight--1", "0");
+        feedbackEditPage.clickAddQuestionButton();
+        feedbackEditPage.waitForTextsForAllStatusMessagesToUserEquals(Const.StatusMessages.FEEDBACK_QUESTION_ADDED);
+        assertNotNull(BackDoor.getFeedbackQuestion(courseId, feedbackSessionName, 1));
+
+        // verify html page
+        feedbackEditPage.verifyHtmlMainContent("/instructorFeedbackMcqQuestionWeightAddSuccess.html");
```

Comment:
The above test case can be extracted to another method `testMcqWeightsFeature_shouldHaveCorrectDefaultValue`.

Reply:
You can separate your test also.
`testMcqWeightsFeature_shouldValidateFieldCorrectly`
`testMcqWeightsFeature_shouldAddQuestionCorrectly` - one failure one success should be enough
`testMcqWeightsFeature_shouldEditQuestionCorrectly` - one failure one success should be enough
`testMcqWeightsFeature_shouldToggleStateCorrectly`

Reply:
@xpdavid Actually, by this I thought you meant that I should extract my existing tests to separate methods and run them all together by running a single `@Test` named `testMcqWeightFeature`. So I did that. But as expected this way the tests are all still interdependent.

so, just to clarify Did you mean that only, or by separating the tests you meant separating them completely and putting them in separate `@Test` functions so that they are not dependent on one another?

Reply:
> putting them in separate @Test functions so that they are not dependent on one another?
If you can do that, it would be great.